compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/FoldConstantLowering.kt

```diff
@@ -0,0 +1,385 @@
+            registerBinaryOp<Long, Long>(LONG, LONG, "or") { a, b -> a.or(b) }
+            registerBinaryOp<Long, Long>(LONG, LONG, "xor") { a, b -> a.xor(b) }
+
+            // String
+            registerUnaryOp<String>(STRING, "toString") { it }
+            registerUnaryOp<String>(STRING, "length") { it.length }
+            registerBinaryOp<String, String>(STRING, STRING, "plus") { a, b -> a.plus(b) }
+            registerBinaryOp<String, String>(STRING, STRING, "compareTo") { a, b -> a.compareTo(b) }
+
+            // Boolean
+            registerUnaryOp<Boolean>(BOOLEAN, "toString") { it.toString() }
+            registerUnaryOp<Boolean>(BOOLEAN, "not") { it.not() }
+            registerBinaryOp<Boolean, Boolean>(BOOLEAN, BOOLEAN, "and") { a, b -> a.and(b) }
+            registerBinaryOp<Boolean, Boolean>(BOOLEAN, BOOLEAN, "compareTo") { a, b -> a.compareTo(b) }
+            registerBinaryOp<Boolean, Boolean>(BOOLEAN, BOOLEAN, "or") { a, b -> a.or(b) }
+            registerBinaryOp<Boolean, Boolean>(BOOLEAN, BOOLEAN, "xor") { a, b -> a.xor(b) }
+
+            // Char
+            registerUnaryOp<Char>(CHAR, "toChar") { it }
+            registerUnaryOp<Char>(CHAR, "toByte") { it.toByte() }
+            registerUnaryOp<Char>(CHAR, "toShort") { it.toShort() }
+            registerUnaryOp<Char>(CHAR, "toInt") { it.toInt() }
+            registerUnaryOp<Char>(CHAR, "toLong") { it.toLong() }
+            registerUnaryOp<Char>(CHAR, "toFloat") { it.toFloat() }
+            registerUnaryOp<Char>(CHAR, "toDouble") { it.toDouble() }
+            registerUnaryOp<Char>(CHAR, "toString") { it.toString() }
+            registerBinaryOp<Char, Char>(CHAR, CHAR, "compareTo") { a, b -> a.compareTo(b) }
+            registerBinaryOp<Char, Char>(CHAR, CHAR, "minus") { a, b -> a.minus(b) }
+            registerBinaryOp<Char, Int>(CHAR, INT, "minus") { a, b -> a.minus(b) }
+            registerBinaryOp<Char, Int>(CHAR, INT, "plus") { a, b -> a.plus(b) }
+
+            // ANY
+            for (type in ANY) {
+                registerBinaryOp<String, Any>(STRING, type, "plus") { a, b -> a.plus(b) }
+                registerBinaryOp<Boolean, Any>(BOOLEAN, type, "equals") { a, b -> a.equals(b) }
+                registerBinaryOp<Char, Any>(CHAR, type, "equals") { a, b -> a.equals(b) }
+                registerBinaryOp<Byte, Any>(BYTE, type, "equals") { a, b -> a.equals(b) }
+                registerBinaryOp<Short, Any>(SHORT, type, "equals") { a, b -> a.equals(b) }
+                registerBinaryOp<Int, Any>(INT, type, "equals") { a, b -> a.equals(b) }
+                registerBinaryOp<Long, Any>(LONG, type, "equals") { a, b -> a.equals(b) }
+                registerBinaryOp<Float, Any>(FLOAT, type, "equals") { a, b -> a.equals(b) }
+                registerBinaryOp<Double, Any>(DOUBLE, type, "equals") { a, b -> a.equals(b) }
+                registerBinaryOp<String, Any>(STRING, type, "equals") { a, b -> a.equals(b) }
+            }
+
+            // IrBuiltins
```

Comment:
It would be awesome to extract names used here into constants in IrBuiltIns and use them where the corresponding operators are created, to be able to quickly see that both refer to the same concept

Reply:
Done.